azure/Guidelines.md

```diff
@@ -747,18 +747,70 @@ The ability to retry failed requests for which a client never received a respons
 <!-- Open API Spec -->
 [OpenAPI Specification]: https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md
 
-### Long Running Operations & Jobs
+### Long-Running Operations & Jobs
 
-Azure generally follows the [Microsoft REST API guidelines for Long running operations](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#13-long-running-operations).  Follow the Microsoft guidelines with the modifications / extensions given here when designing / implementing long running operations.
+When the processing for an operation may take a significant amount of time to complete, it should be
+implemented as a _long-running operation (LRO)_. This allows clients to continue running while the
+operation is being processed. The client obtains the outcome of the operation at some later time
+through another API call.
 
-Previous Azure guidelines specified "Azure-AsyncOperation" as the name of the response header containing the operation URL,
-while the Microsoft guidelines use the name "Operation-Location".
+:white_check_mark: **DO** implement an operation as an LRO if the 99th percentile response time is greater than 1s.
 
-:white_check_mark: **DO** support both `Azure-AsyncOperation` and `Operation-Location` HEADERS, even though they are redundant so that existing SDKs and clients will continue to operate.
+There are two basic patterns that can be used for long-running operations:
+1. Resource-based long-running operations (RELO)
+2. Stepwise long-running operations
+
+:white_check_mark: **DO** use the RELO pattern when the operation is on a resource that contains a "status" property that can be used to obtain the outcome of the operation.
+
+:ballot_box_with_check: **YOU SHOULD** only use the stepwise LRO pattern when the RELO pattern is not applicable.
+
+#### Resource-based long-running operations
+
+Some common situations where the RELO pattern should be used:
+1. A "create" operation for a resource where the basic structure of the resource is created immediately and includes a status field that indicates when the create has completed, e.g. "provisioning" -> "active".
+2. An action operation for a resource where both the initiation of the action and the completion of the action cause a change to the "status" property of the resource.
```

Comment:
If the multiple actions of the same type were requested -- e.g. reboot a VM -- then the service could simply "piggyback" duplicate actions on the first one.  Callers would see the first action complete and take that to mean that "their" action completed.

In other cases, either the service would have to reject actions that were inconsistent with the current outstanding action or the service should use the "stepwise" LRO pattern so that each user could see the outcome of their specific operation.